src/BMPEncoder.ts

```diff
@@ -1,16 +1,31 @@
-'use strict';
+import { IOBuffer } from 'iobuffer';
 
-const { IOBuffer } = require('iobuffer');
+import { BITMAPV5HEADER } from './constants';
 
-const constants = require('./constants');
+export interface DataToEncode {
+  bitDepth: number;
+  height: number;
+  width: number;
+  data:
+    | (number | IOBuffer | ArrayBufferLike | ArrayBufferView | Buffer)
```

Comment:
It doesn't make sense to pass `number`. It would initialize the data to encode with all zeroes.